File: daily/Linked List/2181-Merge-nodes-in-between-zeros.py
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return None

        dummy = ListNode(0)
        curr = dummy

        zero_count = 0
        sum_nodes = 0
        while head:
            if head.val == 0:
                zero_count += 1 
            sum_nodes += head.val
            
            if zero_count >= 2:
                newNode = ListNode(sum_nodes) if sum_nodes > 0 else None
                if newNode:
                    curr.next = newNode
                    curr = curr.next
                zero_count = 1
                sum_nodes = 0

            head = head.next
        return dummy.next

File: daily/Linked List/2181-Merge-nodes-in-between-zeros.go
package daily

func mergeNodes(head *ListNode) *ListNode {
	dummy := &ListNode{Val: 0}
	curr := dummy

	zeroCount := 0
	sumNodes := 0
	for head != nil {
		if head.Val == 0 {
			zeroCount++
		}
		sumNodes += head.Val

		if zeroCount >= 2 {
			newNode := &ListNode{Val: sumNodes}
			if sumNodes > 0 {
				curr.Next = newNode
				curr = curr.Next
			}
			zeroCount = 1
			sumNodes = 0
		}

		head = head.Next
	}
	return dummy.Next
}
